### PR TITLE
feat(client): Stop passing 400 page error messages via query parameters.

### LINKED ISSUE #3649: Stop sending error messages to the 400/500 pages
We initially did this because we did not have Sentry and needed to log errors another way. Now we have Sentry. We should use it, and stop using the `message` query parameter to pass data. Arbitrary *text* can be sent in message, which is displayed to the user. There is no HTML injection attack, but the text field can be used to cause confusion.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Use a session cookie instead.

fixes #3649

@vladikoff and @jrgm - r?

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Tisk tisk, I didn't test this with production resources and Circle is complaining.

**Comment by shane-tomlinson** [comment]
@vladikoff - I'm calling this ready for review - hopefully the tests pass this time.

**Comment by shane-tomlinson** [comment]
@vladikoff - This approach still suffers from https://github.com/mozilla/fxa-content-server/issues/2929. I could remove the session cookie clearing, or set a short expiry, users should probably only see the 400 page once anyways.

**Comment by vladikoff** [comment]
related: https://github.com/mozilla/fxa-content-server/issues/3434

**Comment by shane-tomlinson** [comment]
Failing functional test, pulling back to me.

**Comment by shane-tomlinson** [comment]
@vladikoff - suggestions on handling #2929 and #3434? They seem to conflict. On page reload, we could keep showing the same message, or we could remove the cookie, check for the existence of the cookie after page reload, and if it's not there send the user to `/signin`?

**Comment by vladikoff** [comment]
>  On page reload, we could keep showing the same message, or we could remove the cookie

I think we should remove the cookie in the hope the refresh fixed the problem (if js error)

**Comment by vladikoff** [comment]
from mtg: watch out for the spinner, it hides things!

**Comment by vladikoff** [comment]
from mtg: stop redirecting, replace the current view with the whatever the 400.html shows now:



User refreshes, either it fixes itself or shows the same error. That should also fix issue: https://github.com/mozilla/fxa-content-server/issues/2929 // https://github.com/mozilla/fxa-content-server/issues/3434

**Comment by vladikoff** [comment]
> #2929 // #3434

wow lucky numbers. from mtg: We also should do the same for `unexpected_error` and `500.html`

**Comment by shane-tomlinson** [comment]
@vladikoff - I have updated this with the changes we talked about on Friday. Passing over to you!

**Comment by vladikoff** [comment]
Needs rebase

**Comment by vladikoff** [comment]
@shane-tomlinson one issue I found, if you navigate directly to `http://localhost:3030/unexpected_error` then it spins forever

File: app/scripts/lib/constants.js [hunk 1]
```diff
@@ -78,12 +78,6 @@ define(function (require, exports, module) {
     PROFILE_IMAGE_MIN_WIDTH: 100,
     DEFAULT_PROFILE_IMAGE_MIME_TYPE: 'image/jpeg',
 
-    INTERNAL_ERROR_PAGE: '/500.html',
-    BAD_REQUEST_PAGE: '/400.html',
-    // delay before redirecting to the error page to
-    // ensure metrics are reported to the backend.
-    ERROR_REDIRECT_TIMEOUT_MS: 1000,
```
Comment (vladikoff) [inline app/scripts/lib/constants.js]: 👍

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson one issue I found, if you navigate directly to http://localhost:3030/unexpected_error then it spins forever

Did you restart your content server? This is what I see:

screen shot 2016-05-25 at 10 37 40

**Comment by shane-tomlinson** [comment]
@vladikoff - rebased.

**Pull request merged**